apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainProcessorImpl.java

```diff
@@ -1100,25 +1109,37 @@ private Block popLastBlock() {
     private void popOffWithRescan(int height) {
         globalSync.writeLock();
         try {
+            int scanHeight = 0;
+            int shardInitialHeight = blockchain.getShardInitialBlock().getHeight();
+            if ( shardInitialHeight > 0 ) {
+                scanHeight = Math.max(height, shardInitialHeight);
+            }
+            log.debug("Set the height for Scan process to {}, shards' initialBlock height={}", scanHeight, shardInitialHeight);
             try {
-                scheduleScan(0, false);
+                scheduleScan(scanHeight, false);
                 long blockIdAtHeight = blockchain.getBlockIdAtHeight(height);
                 Block lastBLock = blockchain.deleteBlocksFrom(blockIdAtHeight);
                 // rollback not scan safe derived tables with blocks and transactions
                 for (DerivedTableInterface derivedTable : dbTables.getDerivedTables()) {
                     if (!derivedTable.isScanSafe()) {
+                        log.debug("Rollback not scan safe table {}", derivedTable.getName());
                         derivedTable.rollback(height);
                     }
                 }
                 lookupBlockhain().setLastBlock(lastBLock);
-
                 lookupBlockhainConfigUpdater().rollback(lastBLock.getHeight());
-                log.debug("Deleted blocks starting from height {}", height);
+                log.debug("Blockchain config updated, lastblock={} at height={}", lastBLock.getId(), lastBLock.getHeight());
             } finally {
-                scan(0, false);
+                try {
+                    scan(scanHeight, false);
+                }catch (BlockchainScanException e){
+                    log.error("CRITICAL ERROR. PLEASE REPORT TO THE DEVELOPERS.\n" + e.toString(), e);
+                    System.exit(-1);
```

Comment:
Looks like its not an expected behavior.

Reply:
At this moment it's a predictable behavior with sharded Db.

Reply:
removed